osu.Game/Users/UserGridPanel.cs

```diff
@@ -4,139 +4,232 @@
+                            new Dimension()
                         },
-                        new Drawable[]
+                        RowDimensions = new[] { new Dimension() },
+                        Content = new[]
                         {
-                            CreateStatusIcon().With(icon =>
-                            {
-                                icon.Anchor = Anchor.Centre;
-                                icon.Origin = Anchor.Centre;
-                            }),
-                            CreateStatusMessage(false).With(message =>
+                            new Drawable[]
                             {
-                                message.Anchor = Anchor.CentreLeft;
-                                message.Origin = Anchor.CentreLeft;
-                                message.Margin = new MarginPadding { Left = margin };
-                            })
+                                CreateStatusIcon().With(icon =>
+                                {
+                                    icon.Anchor = Anchor.Centre;
+                                    icon.Origin = Anchor.Centre;
+                                    icon.Size = new Vector2(14);
+                                }),
+                                CreateStatusMessage(false).With(message =>
+                                {
+                                    message.Anchor = Anchor.CentreLeft;
+                                    message.Origin = Anchor.CentreLeft;
+                                    message.Margin = new MarginPadding { Left = 18 };
+                                })
+                            }
                         }
                     }
                 }
             };
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            var ruleset = rulesets?.GetRuleset(User.PlayMode);
+
+            level.User.Value = new UserProfileData(User, ruleset.AsNonNull());
+
+            // some API calls don't return PlayMode - we don't want to draw a ruleset icon in that case
+            if (ruleset != null)
+            {
+                details.Add(ruleset.CreateInstance().CreateIcon().With(
+                    icon =>
+                    {
+                        icon.Size = new Vector2(16);
+                        icon.Margin = new MarginPadding { Right = 8 };
+                    }));
+            }
+
+            details.Add(new GroupBadgeFlow
+            {
+                User = { Value = User }
+            });
 
             if (User.IsSupporter)
             {
                 details.Add(new SupporterIcon
                 {
-                    Height = 26,
+                    Height = 16,
                     SupportLevel = User.SupportLevel
                 });
             }
+        }
+
+        protected override bool OnHover(HoverEvent e)
+        {
+            hoverDim?.FadeTo(0.2f, 200);
```

Comment:
You might wanna put an easing on these, it looks pretty bad right now. `OutQuint` is what most of everything uses.